### PR TITLE
fix #75048 trans.firm.in

### LINKED ISSUE #75048: trans.firm.in
### Issue URL (Ads)
 [https://trans.firm.in/img-601a61fb799d1.html](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Ftrans.firm.in%2Fimg-601a61fb799d1.html)

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>

### System configuration

Information | value
--- | ---
AdGuard product: | AdGuard for Windows v7.6 nightly 17
Browser: | Opera
AdGuard driver: | WFP
Stealth mode options: | Hide your search queries, <br>Send Do-Not-Track header, <br>Remove X-Client-Data header from HTTP requests, <br>Strip URLs from tracking parameters, <br>Self-destructing third-party cookies (180), <br>Block Push API, <br>Block Location API
DNS filtering: | server: `https://dns-unfiltered.adguard.com/dns-query`<br>filters: `User rules`
Filters: | <b>Ad Blocking:</b><br/> EasyList, <br/>Peter Lowe's Blocklist, <br/>AdGuard Mobile Ads, <br/>AdGuard Base<br/><br/><b>Privacy:</b><br/> EasyPrivacy, <br/>AdGuard Tracking Protection<br/><br/><b>Social Widgets:</b><br/> AdGuard Social Media<br/><br/><b>Annoyances:</b><br/> Adblock Warning Removal List, <br/>I don't care about cookies, <br/>EasyList Cookie List, <br/>AdGuard Annoyances<br/><br/><b>Security:</b><br/> Online Malicious URL Blocklist, <br/>Spam404, <br/>NoCoin Filter List<br/><br/><b>Language-specific:</b><br/> AdGuard Turkish, <br/>AdGuard Russian
Custom filter subscriptions: | https://raw.githubusercontent.com/AdguardTeam/FiltersRegistry/master/filters/filter_15_DnsFilter/filter.txt

### PR DESCRIPTION
**Pull request by wout-codes**
#75048

### CODE REVIEW

**Comment by wout-codes** [comment]
@zloyden Please take a look at this.

File: EnglishFilter/sections/general_extensions.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ shameless.com#%#//scriptlet("abort-current-inline-script", "$LAB", "/getexoloade
 !
 ! 'PopJSLock'
 ! #%#//scriptlet("abort-current-inline-script", "String.fromCharCode", "constructor")
-freepornxxxhd.com,givemenflstreams.com,givemenbastreams.com,givememmastreams.com,givemeredditstreams.com,shinden.pl,ozporno.com,area51deportiva.com,xxxfree.watch,linksly.co,animesanka.com,mhometheater.com#%#//scriptlet("abort-current-inline-script", "String.fromCharCode", "constructor")
+tmnlsmvgrkoj.com,xokikdassjjhrn.com,freepornxxxhd.com,givemenflstreams.com,givemenbastreams.com,givememmastreams.com,givemeredditstreams.com,shinden.pl,ozporno.com,area51deportiva.com,xxxfree.watch,linksly.co,animesanka.com,mhometheater.com#%#//scriptlet("abort-current-inline-script", "String.fromCharCode", "constructor")
```
Comment (zloyden) [inline EnglishFilter/sections/general_extensions.txt:30]: Why did you add those domains here?

File: EnglishFilter/sections/general_extensions.txt [hunk 2]
```diff
@@ -98,7 +98,7 @@ gaybeeg.info,italiashare.life#%#(function(){var b=window.setTimeout;window.setTi
...
+glygrebedu.pro,autoizers.*,vizplay.*,view47.*,streamplusvip.*,oyohd.com,hindilinks4uto.com,animestream.watch,msubplix.com,vidxhot.net,waaw.tv,waaw.to,chillicams.net,flashx.net,hqq.to,justswallows.com,openload.casa,meucdn.vip,yandexcdn.com,firstr0w.eu,hqq.tv,onlystream.tv,nhentai.net,streamz.*,hindipix.in,3movs.com,alltube.tv,oload.monster,tryboobs.com,eporner.com,oload.press,vivatube.com,oload.life,streamp1ay.me,nxload.com,oload.services,verystream.com,freeteenporn.xxx,ashemale.one,kaotic.com,hentaicore.org,oload.vip,woof.tube,vidop.icu,oload.best,oload.network,alltube.pl,oload.biz,oload.website,oload.xyz#%#//scriptlet("abort-on-property-read", "BetterJsPop")
```
Comment (zloyden) [inline EnglishFilter/sections/general_extensions.txt:101]: The same question.

File: EnglishFilter/sections/general_extensions.txt [hunk 3]
```diff
@@ -157,9 +157,9 @@ imgdawgknuttz.com,getlec.blogspot.com#%#//scriptlet("abort-on-property-read", "z
...
...ack-games.com,apkshrt.com,linkshorts.me,fusionmovies.to,lefturl.com,gamepciso.com,100count.net,gocoo.co,hdss.video,playerbiasa.xyz,akwams.*,freelitecoin.vip,freebitcoin.vip,bigb0ss.net,racaty.net,videobin.co,bdupload.asia,indishare.org,gdtot.com,livestreaming24.eu,ilinks.in,hdmovieplus.site,egy.best,egybest.*,upvid.net,filmvilag.me,revivelink.com,owllink.net,goo-2o.com,akwam.*,multicanais.com,upstream.to,streamtape.com,streamingcommunity.to,easyload.io,clipconverter.cc,readm.org,notube.net,gogo-stream.com,animesonline.cz,doctor-groups.com,kissanime.nz,gowatchseries.ac,aatfal.com,ouo.io,budcenter.site,ustreamix.to,ffmovies.co,animekisa.tv,shrinkbtc.cc,wlnk.ec,zone-annuaire.best,repackov.com,dood.*,youdbox.com,databasegdriveplayer.xyz,putlocker.style,watchmovies7.com.pk,manhwa18.com,swatchseries.to,mangatx.com,ouofly.com,playtube.ws,iir.ai,bolssc.com,linkjust1.com,fx4vip.com,mangazuki.online,elsfile.org,clicknupload.co,cima4u.io,mcloud.to,fmovies.to,aflamgulf.com,gateaflam.com,exee.io,anonymous-links.com,mediumarticles.com,pulsemens.com,aparat.cam,doramasmp4.com,jkanime.net,flvto.biz,movs4u.live,lookmovie.ag$$script[tag-content="break;case $."][min-length="30000"][max-length="85000"]
```
Comment (zloyden) [inline EnglishFilter/sections/general_extensions.txt:160]: Why not `trans.firm.in`?

File: EnglishFilter/sections/general_extensions.txt [hunk 4]
```diff
@@ -157,9 +157,9 @@ imgdawgknuttz.com,getlec.blogspot.com#%#//scriptlet("abort-on-property-read", "z
...
...nkshorts.me,fusionmovies.to,lefturl.com,gamepciso.com,100count.net,gocoo.co,hdss.video,playerbiasa.xyz,akwams.*,freelitecoin.vip,freebitcoin.vip,bigb0ss.net,racaty.net,videobin.co,bdupload.asia,indishare.org,gdtot.com,livestreaming24.eu,ilinks.in,hdmovieplus.site,egy.best,egybest.*,upvid.net,filmvilag.me,bitfly.io,revivelink.com,owllink.net,goo-2o.com,akwam.*,multicanais.com,upstream.to,streamingcommunity.to,easyload.io,clipconverter.cc,readm.org,notube.net,gogo-stream.com,animesonline.cz,doctor-groups.com,kissanime.nz,gowatchseries.ac,aatfal.com,ouo.io,budcenter.site,ustreamix.to,ffmovies.co,animekisa.tv,shrinkbtc.cc,wlnk.ec,zone-annuaire.best,repackov.com,dood.*,youdbox.com,databasegdriveplayer.xyz,putlocker.style,watchmovies7.com.pk,manhwa18.com,swatchseries.to,mangatx.com,ouofly.com,playtube.ws,iir.ai,bolssc.com,linkjust1.com,fx4vip.com,mangazuki.online,elsfile.org,clicknupload.co,cima4u.io,mcloud.to,fmovies.to,aflamgulf.com,gateaflam.com,exee.io,anonymous-links.com,mediumarticles.com,pulsemens.com,aparat.cam,doramasmp4.com,jkanime.net,pcgamez-download.com,flvto.biz,movs4u.live#%#//scriptlet("abort-current-inline-script", "String.fromCharCode", "/emZnbG9hZGVk|break;case \$\./")
```
Comment (zloyden) [inline EnglishFilter/sections/general_extensions.txt:162]: The same question about `trans.firm.in`.

File: RussianFilter/sections/adservers.txt [hunk 5]
```diff
@@ -5,6 +5,10 @@
 ! `window.adswellBaseDomain`
 ! https://damskiy.xyz/vendor/lib/oY.js
 !
+||xokikdassjjhrn.com^$document,popup
+||tmnlsmvgrkoj.com^$document,popup
+||glygrebedu.pro^
+||tmnlsmvgrkoj.com^
```
Comment (zloyden) [inline RussianFilter/sections/adservers.txt:11]: I guess those ones are adservers. Why you didn't use `$third-party`?
Also missed `||autoizers.info^$third-party`.

**Review by zloyden — CHANGES_REQUESTED** [review]

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:30]: Check yourself why. Too much too explain.

File: EnglishFilter/sections/general_extensions.txt (hunk 2 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:101]: As I said, check yourself, too much to explain

**Review by zloyden — CHANGES_REQUESTED** [review]
There is an adb detection that called `absda`. Add `trans.firm.in` to the common rules.

File: EnglishFilter/sections/general_extensions.txt (hunk 3 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:160]: Ok. I agree.

**Comment by wout-codes** [comment]
Adb detection not reproduced with russian ip

File: RussianFilter/sections/adservers.txt (hunk 5 above)
Comment (wout-codes) [reply RussianFilter/sections/adservers.txt:11]: Ok, I agree.

**Comment by zloyden** [comment]
> Adb detection not reproduced with russian ip

<details><summary>Screenshot:</summary>

image

</details><br/>

**Comment by wout-codes** [comment]
Ok wait, let me read the code and check..

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (zloyden) [reply EnglishFilter/sections/general_extensions.txt:30]: > Check yourself why. Too much too explain.

Be prepared for the fact that our team can ask you questions. This is not proof. I can't approve it without an explanation.

File: EnglishFilter/sections/general_extensions.txt (hunk 2 above)
Comment (zloyden) [reply EnglishFilter/sections/general_extensions.txt:101]: [The same](https://github.com/AdguardTeam/AdguardFilters/pull/75051#discussion_r577474504).

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:30]: Ok wait.. Let me send a screenshot.

Comment (zloyden) [reply EnglishFilter/sections/general_extensions.txt:30]: This scriptlet is for original domains, not the adservers.

Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:30]: image
@zloyden

File: EnglishFilter/sections/general_extensions.txt (hunk 2 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:101]: @zloyden 
image

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt:30]: Ah yes, those adservers does the job. Quite forgot that.

File: EnglishFilter/sections/general_extensions.txt (hunk 2 above)
Comment (zloyden) [reply EnglishFilter/sections/general_extensions.txt:101]: As I said earlier, this scriptlet is for original domains, not the adservers. Also please use the spoiler for the screenshots.

Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt]: I do not know how. I have removed them.

File: EnglishFilter/sections/general_extensions.txt (hunk 1 above)
Comment (zloyden) [inline EnglishFilter/sections/general_extensions.txt]: So, the general idea about this one, you add the original domain here (not the adserver).
But, you handle it via blocking rules, including Regex. No need for adding domains to the scriptlet.

Comment (wout-codes) [reply EnglishFilter/sections/general_extensions.txt]: Ok.

File: RussianFilter/sections/adservers.txt [hunk 6]
```diff
@@ -5,6 +5,12 @@
 ! `window.adswellBaseDomain`
 ! https://damskiy.xyz/vendor/lib/oY.js
 !
+||xokikdassjjhrn.com^$document,popup
+||tmnlsmvgrkoj.com^$document,popup
+||autoizers.info^$document,popup
```
Comment (zloyden) [inline RussianFilter/sections/adservers.txt:10]: Just remove it, you've already had `||autoizers.info^$third-party`.

**Review by zloyden — CHANGES_REQUESTED** [review]

File: RussianFilter/sections/adservers.txt (hunk 6 above)
Comment (wout-codes) [reply RussianFilter/sections/adservers.txt:10]: Also for the others?

Comment (wout-codes) [reply RussianFilter/sections/adservers.txt:10]: Can I use `$third-party` for all the domains I added and removing the `$document,popup`? Since it's not supported by the apps.

Comment (zloyden) [reply RussianFilter/sections/adservers.txt:10]: Yes, for others too. After that I think there is nothing to do - looks ok.

Comment (wout-codes) [reply RussianFilter/sections/adservers.txt:10]: Ok done.

**Comment by zloyden** [comment]
Well, let me check.

**Review by zloyden — APPROVED** [review]

File: EnglishFilter/sections/general_extensions.txt (hunk 2 above)
Comment (zloyden) [reply EnglishFilter/sections/general_extensions.txt]: No problem, I edited your post. There is a code for the spoiler.

```
<details><summary>Screenshot:</summary>

![image](%screenshot_url%)
</details><br/>
```
